### DISCUSSION TITLE
Is it possible to delete a Check Run? Or report netural status on a Check Suite w/o a run?

### DISCUSSION THREAD

**Discussion by chrsmith** [community/community#24799] [Apps, API and Webhooks]
<p>I work at a company that allows you to deploy cloud infrastructure using code. And we have a GitHub application that surfaces the changes to your cloud infrastructure that were previewed/applied as part of a pull request. (You can get more context <a href="https://pulumi.io/reference/cd-github.html" rel="nofollow noopener">here</a>.)</p>
<p>The problem is that by the nature of how our system works, <em>we don’t know ahead of time</em> what will actually happen for a given commit. Rather, we rely on the user to have configured their CI/CD system to invoke our command-line tool. And we then surface those results by reporting new GitHub check runs.</p>
<p>This leads to the unfortunate UX where if the commit didn’t end up making any cloud infrastructure changes, the Check Suite associated with our GitHub app doesn’t report any Check Runs. And the GitHub UI just says “Queued XX minutes ago”. (And later “Queued XX days ago”.)</p>
<p>I’d like to improve this if possible, though the API docs indiciate this isn’t really possible. So this is generally feedback on a scenario which doesn’t fit well into the current design.</p>
<p>(1) Is there a way to have the GitHub app immediately report a “neutral” status for the Check Suite, without creating a new Check Run? Rather than being in the “Queued” state, is it possible to report a neutral status immediately?</p>
<p>(2) If not, we can just as easily create a “placeholder” Check Run. With a descriptive title like “waiting for updates”. However, if we do that, is it possible to <em>delete</em> that Check Run later? It doesn’t make sense to show “waiting for updates” along side of the updates that eventually took place.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out.</p>
<p>Here’s my understanding of the Checks flow from <a href="https://developer.github.com/v3/checks/" rel="nofollow noopener">the documentation</a>:</p>
<ol>
<li>Someone pushes code to the repository</li>
<li>GitHub creates a Check Suite object and pushes a <code>check_suite</code> webhook event to your GitHub App</li>
<li>At this point, if your app does nothing you get the “Queued XX YY ago” behavior you’re describing</li>
<li>Your GitHub App creates a Check Run object with the resulting status</li>
<li>The Check Suite object aggregates all of the Check Run objects’ status into a single result</li>
</ol>
<p>What it sounds like you’re currently doing is that when you receive a <code>check_suite</code> webhook event, and your command-line app doesn’t get called, no Check Run object is created and you end up at the failure case of 2i above.</p>
<p>No, there isn’t a way to delete a Check Run object, so the key to your question is what to do in step 3 above. What I would suggest is to immediately create the Check Run object with the <code>in_progress</code> status as described in <a href="https://developer.github.com/v3/checks/runs/" rel="nofollow noopener">the Check Run documentation</a>. Then, when you’ve determined that nothing is supposed to happen, you can <a href="https://developer.github.com/v3/checks/runs/#update-a-check-run" rel="nofollow noopener">update</a> (not delete) that Check Run object with a status of <code>completed</code> and whatever output you want.</p>
<p>I hope that helps!</p>

**Comment by johnsonsamuel** [comment]
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> can you please share how do I update the check run status ? because I want to update the check run status to failure but I get the error </p>
<pre><code>ERROR HttpError: Invalid app_id #appIdNumber
</code></pre>
<p>Method used:</p>
<pre><code>context.github.checks.update({
          owner:'johnsonsamuel',
          repo:'react-app-test',
          check_run_id: context.payload.check_run.id
        })
</code></pre>
<p>Any suggestion ?</p>

**Comment by lee-dohm** [comment]
<p>The error you’re seeing is that you’re supplying an invalid app ID number. This makes me think that you’re doing something wrong during the authentication steps or you’re attempting to manipulate a check run belonging to a different installation or something.</p>
<p>I hope that helps.</p>

**Comment by johnsonsamuel** [comment]
<p>Thanks for the reply. Even I was thinking about the same but I am able to create run without any problems. Only updating a check run is giving me this error. </p>

**Comment by lee-dohm** [comment]
<p>You may want to contact private support at <a href="https://support.github.com/contact" rel="nofollow noopener">https://support.github.com/contact</a> then so that we can take a deeper look at your configuration and exactly what parameters you’re passing to the API calls.</p>

**Comment by johnsonsamuel** [comment]
<p>sure <a class="mention" href="/lee-dohm">@lee-dohm</a> . Will do. Thanks again :slight_smile:</p>
